### ISSUE TITLE
NullReferenceException in MySqlConnection.Cancel 

### ISSUE THREAD

**Issue by dmathijs** [mysql-net/MySqlConnector#1506]
MySqlConnector version: 2.3.1
Server type (MySQL, MariaDB, Aurora, etc.) and version: MariaDb
.NET version: 8.0.302
(Optional) ORM NuGet packages and versions:  Pomelo.EntityFrameworkCore.MySql 

Today I got this stacktrace on my server:
```
Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
   at MySqlConnector.MySqlConnection.Cancel(ICancellableCommand command, Int32 commandId, Boolean isCancel) in /_/src/MySqlConnector/MySqlConnection.cs:line 762
   at MySqlConnector.MySqlCommand.CancelCommandForTimeout() in /_/src/MySqlConnector/MySqlCommand.cs:line 439
   at MySqlConnector.Utilities.TimerQueue.Callback(Object obj) in /_/src/MySqlConnector/Utilities/TimerQueue.cs:line 98
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.TimerQueueTimer.Fire(Boolean isThreadPool)
   at System.Threading.TimerQueue.FireNextTimers()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
```
I noticed that back in 2022, there was a fix done for this exact issue:
- https://github.com/mysql-net/MySqlConnector/issues/1177
- https://github.com/mysql-net/MySqlConnector/pull/1183/commits/eadb98c1d6373dbcd517f2470a3fdf86bdd72113

However, this seems to be re-factored away and it seems like this bug was re-introduced at some point.
https://github.com/mysql-net/MySqlConnector/commit/000f586ce78e7c858b697d6e3b94cfcc68ac3162
https://github.com/mysql-net/MySqlConnector/commit/e82f091e875d74cce89f46982f68f5b1e40fbca8

The problem is that I don't know enough about the code-base to know whether adding null conditional operators is allowed, can someone give me some pointers?

**Comment by bgrainger** [comment]
This does look like a bug that was reintroduced in https://github.com/mysql-net/MySqlConnector/commit/e82f091e875d74cce89f46982f68f5b1e40fbca8; thanks for the analysis.

**Comment by dmathijs** [comment]
Thank you!

**Comment by bgrainger** [comment]
Fixed in [2.4.0](https://github.com/mysql-net/MySqlConnector/releases/tag/2.4.0).
